plan/aggregation_pruning.go

```
@@ -71,35 +57,41 @@ func (ap *aggPruner) eliminateAggregation(p LogicalPlan) (LogicalPlan, error) {
 	}
 	newChildren := make([]Plan, 0, len(p.Children()))
 	for _, child := range p.Children() {
-		newChild, err := ap.eliminateAggregation(child.(LogicalPlan))
-		if err != nil {
-			return nil, errors.Trace(err)
-		}
+		newChild := ap.eliminateAggregation(child.(LogicalPlan))
 		newChildren = append(newChildren, newChild)
 	}
 	retPlan.SetChildren(newChildren...)
-	return retPlan, nil
+	return retPlan
+}
+
+func convertAggToProj(agg *Aggregation, ctx context.Context, allocator *idAllocator) *Projection {
+	proj := &Projection{
+		Exprs:           make([]expression.Expression, 0, len(agg.AggFuncs)),
+		baseLogicalPlan: newBaseLogicalPlan(Proj, allocator),
+	}
+	proj.self = proj
+	proj.initIDAndContext(ctx)
+	for _, fun := range agg.AggFuncs {
+		expr := rewriteExpr(fun.GetArgs(), fun.GetName(), ctx)
+		proj.Exprs = append(proj.Exprs, expr)
+	}
+	proj.SetSchema(agg.schema.Clone())
+	return proj
 }
 
 // rewriteExpr will rewrite the aggregate function to expression doesn't contain aggregate function.
-func (ap *aggPruner) rewriteExpr(exprs []expression.Expression, funcName string) (newExpr expression.Expression, err error) {
+func rewriteExpr(exprs []expression.Expression, funcName string, ctx context.Context) (newExpr expression.Expression) {
 	switch funcName {
 	case ast.AggFuncCount:
 		// If is count(expr), we will change it to if(isnull(expr), 0, 1).
 		// If is count(distinct x, y, z) we will change it to if(isnull(x) or isnull(y) or isnull(z), 0, 1).
 		isNullExprs := make([]expression.Expression, 0, len(exprs))
 		for _, expr := range exprs {
-			isNullExpr, err := expression.NewFunction(ap.ctx, ast.IsNull, types.NewFieldType(mysql.TypeTiny), expr.Clone())
-			if err != nil {
-				return nil, errors.Trace(err)
-			}
+			isNullExpr, _ := expression.NewFunction(ctx, ast.IsNull, types.NewFieldType(mysql.TypeTiny), expr.Clone())
```

Why ignore the error?